pallets/protos/src/lib.rs

```diff
@@ -134,11 +137,10 @@ pub mod pallet {
 		StorageMap<_, Identity, Hash256, Proto<T::AccountId, T::BlockNumber>>;
 
 	#[pallet::storage]
-	pub type ProtosByTag<T: Config> = StorageMap<_, Blake2_128Concat, Tags, Vec<Hash256>>;
+	pub type ProtosByTag<T: Config> = StorageDoubleMap<_, Blake2_128Concat, Tags, Blake2_128Concat, Hash256, bool>;
```

Comment:
`Hash256` is already an hash so we can use `Identity` instead of `Blake2_128Concat` for `Hash256` types